src/lucky_cli/browser_authentication_src_template.cr

```diff
@@ -1,3 +1,167 @@
-class BrowserAuthenticationSrcTemplate < Teeplate::FileTree
-  directory "#{__DIR__}/../browser_authentication_app_skeleton"
+class BrowserAuthenticationSrcTemplate
+  def render(path : Path)
+    LuckyTemplate.write!(path, template_folder)
+  end
+
+  def template_folder
+    LuckyTemplate.create_folder do |root_dir|
+      root_dir.add_folder("spec") do |spec_dir|
+        spec_dir.add_folder("flows") do |flows_dir|
+          flows_dir.add_file("authentication_spec.cr") do |io|
+            ECR.embed("#{__DIR__}/../browser_authentication_app_skeleton/spec/flows/authentication_spec.cr.ecr", io)
+          end
+          flows_dir.add_file("reset_password_spec.cr") do |io|
+            ECR.embed("#{__DIR__}/../browser_authentication_app_skeleton/spec/flows/reset_password_spec.cr.ecr", io)
+          end
+        end
+        spec_dir.add_folder("support") do |support_dir|
+          support_dir.add_file(".keep")
+          support_dir.add_folder("flows") do |flows_dir|
+            flows_dir.add_file("authentication_flow.cr") do |io|
+              ECR.embed("#{__DIR__}/../browser_authentication_app_skeleton/spec/support/flows/authentication_flow.cr.ecr", io)
+            end
+            flows_dir.add_file("reset_password_flow.cr") do |io|
+              ECR.embed("#{__DIR__}/../browser_authentication_app_skeleton/spec/support/flows/reset_password_flow.cr.ecr", io)
+            end
+          end
+        end
+      end
+      root_dir.add_folder("src") do |src_dir|
+        src_dir.insert_folder("actions", actions_folder)
+        src_dir.insert_folder("emails", emails_folder)
+        src_dir.insert_folder("pages", pages_folder)
+      end
+    end
+  end
+
+  private def actions_folder
+    LuckyTemplate.create_folder do |actions_dir|
+      actions_dir.add_folder("me") do |me_dir|
+        me_dir.add_file("show.cr") do |io|
+          ECR.embed("#{__DIR__}/../browser_authentication_app_skeleton/src/actions/me/show.cr.ecr", io)
+        end
+      end
+      actions_dir.add_folder("mixins") do |mixins_dir|
+        mixins_dir.add_file(".keep")
```

Comment:
Maybe we can knock out https://github.com/luckyframework/lucky_cli/issues/778 too while you're in here? This issue is basically saying no `.keep` inside folders that will always have files generated. We can also come back to that later once everything is in place.

Reply:
The only issue I have removing `.keep` files are that it removes the _intention_ of whether a particular folder structure should _always exist_, e.g. `mixins` is a folder name you _expect_ to see in a project. Or `support` is a folder name you _expect_ to see in `spec`. That's my opinion, otherwise I agree you don't need to include them (although, they don't cause any harm in having them).

Reply:
We could always add an option to the CLI that doesn't add those files, which seems like the better option here.

Reply:
I'm cool with that too. The files being there don't really bother me. We can just leave it for now and worry about that in a future PR.